File: transposition-rs/railfence-rs/rail_fence.rs
pub fn encrypt(plain_text: &String, depth: &usize) -> String {
    /* TODO - Find a math formula if possible */

    let mut fence: Vec<Vec<char>> = init_fence(depth);

    let mut i = 0;
    let mut going_down = true;
    for c in plain_text.to_uppercase().chars() {
        if c < 'A' || c > 'Z' {
            continue;
        }

        fence[i].push(c);

        if (i == 0 && !going_down) || (i == depth-1 && going_down) {
            going_down = !going_down;
        }

        if going_down {
            i += 1;
        } else {
            i -= 1;
        }
    }

    let mut cipher_text = String::new();
    for rail in fence {
        for c in rail {
            cipher_text.push(c);
        }
    }
    cipher_text
}

pub fn decrypt(cipher_text: &String, depth: &usize) -> String {
    let mut plain_text = String::from("Plain");

    plain_text
}

fn init_fence(depth: &usize) -> Vec<Vec<char>> {
    let mut fence: Vec<Vec<char>> = Vec::new();

    for i in 0..*depth {
        let mut tmp: Vec<char> = Vec::new();
        fence.push(tmp);
    }

    fence
}

File: python/rail_fence.py
def main():
	pass

def encrypt(plain_text, depth):
	plain_text = plain_text.replace(" ", "")
	plain_text = plain_text.upper()

	rails = [""] * depth
	d = 0
	to_fro_ptr = 0
	going_down = True
	for c in plain_text:
		rails[to_fro_ptr] += c
		if (to_fro_ptr == 0 and not going_down) or (to_fro_ptr == depth -1 and going_down):
			going_down = not going_down

		to_fro_ptr += 1 if going_down else -1
	
	print(rails)
	cipher_text = "".join(rails)
	return cipher_text

def decrypt(cipher_text, depth):
	cipher_text = cipher_text.replace(" ", "")
	cipher_text = cipher_text.upper()

	rails = [[""] * len(cipher_text) for _ in range(depth)]
	text_ptr = 0
	for i in range(len(rails)):
		to_fro_ptr = 0
		going_down = True
		for j in range(len(cipher_text)):
			if to_fro_ptr == i:
				rails[i][j] = cipher_text[text_ptr]
				text_ptr += 1
			if (to_fro_ptr == 0 and not going_down) or (to_fro_ptr == depth -1 and going_down):
				going_down = not going_down

			to_fro_ptr += 1 if going_down else -1
	
	to_fro_ptr = 0
	going_down = True
	plain_text = ""
	for j in range(len(cipher_text)):
		#print(to_fro_ptr, j)
		plain_text += rails[to_fro_ptr][j]			
		if (to_fro_ptr == 0 and not going_down) or (to_fro_ptr == depth -1 and going_down):
			going_down = not going_down

		to_fro_ptr += 1 if going_down else -1
	
	return plain_text

if __name__ == "__main__":
	main()
